src/modules/audio/audio.c

```diff
@@ -108,32 +117,37 @@ static void onPlayback(ma_device* device, void* output, const void* _, uint32_t
     }
   }
 
-  ma_mutex_unlock(&state.locks[0]);
+  ma_mutex_unlock(&state.locks[AUDIO_PLAYBACK]);
 }
 
 static void onCapture(ma_device* device, void* output, const void* input, uint32_t frames) {
-  ma_mutex_lock(&state.locks[1]);
-  ma_mutex_unlock(&state.locks[1]);
+  // note: ma_pcm_rb is lockless
+  void *store;
+  while(frames > 0) {
+    uint32_t available_frames = frames;
+    ma_result acquire_status = ma_pcm_rb_acquire_write(&state.capture_ringbuffer, &available_frames, &store);
+    if(acquire_status != MA_SUCCESS) {
+      fprintf(stderr, "Dropping mic audio, failed to acquire ring buffer: %d\n", acquire_status);
```

Comment:
Would like to try to use `lovrAssert` or `lovrLog` here if possible.  Calling those on threads isn't possible, but it may be that some sort of `captureBroken` flag could be set in state that `lovr.audio.capture` checks?  Not sure.